visualizer/src/service/tools/segmentMachine.js

```diff
@@ -198,16 +143,36 @@ const segmentMachine = Machine(
   {
     guards: {
       ...toolGuards,
-      usingColorTool: ({ tool }) => colorTools.includes(tool),
+      usingColorTool: ({ tool }) => ['brush', 'select', 'trim', 'flood'].includes(tool),
+      isColorTool: (_, { tool }) => ['brush', 'select', 'trim', 'flood'].includes(tool),
+      usingGrayscaleTool: ({ tool }) =>
+        ['brush', 'select', 'trim', 'flood', 'threshold', 'watershed'].includes(tool),
+      isGrayscaleTool: (_, { tool }) =>
+        ['brush', 'select', 'trim', 'flood', 'threshold', 'watershed'].includes(tool),
+      isNoPanTool: (_, { tool }) => ['brush', 'threshold'].includes(tool),
+      isPanTool: (_, { tool }) => ['select', 'trim', 'flood', 'watershed'].includes(tool),
```

Comment:
Not wild about all these lists, but not a blocker for this PR. Would be nice to classify these in a more clear way.

Reply:
I'll put them into const lists at the top of the file and reference them here